src/test/java/org/reaktivity/nukleus/tls/internal/streams/ServerIT.java

```diff
@@ -70,7 +80,97 @@ public void shouldEstablishConnection() throws Exception
     @ScriptProperty({
             "newClientAcceptRef ${newServerConnectRef}",
             "clientAccept \"nukleus://target/streams/tls#source\"" })
-    public void shouldEstablishConnectionWithAlpn() throws Exception
+    public void shouldNegotiateWithALPN() throws Exception
+    {
+        k3po.finish();
+    }
+
+    /*
+     * no route for protocol2, route for null protocol
+     * client sends ALPN w/ protocol2
+     * negotiates successfully without ALPN, so protocol == "" => null    --> BEGIN(appplicationProtocol=null)
+     */
+    @Test
+    @Specification({
+            "${route}/server/controller",
+            "${client}/connection.established.with.alpn/client",
+            "${server}/connection.established/server" })
+    @ScriptProperty({
+            "newClientAcceptRef ${newServerConnectRef}",
+            "clientAccept \"nukleus://target/streams/tls#source\"" })
+    public void shouldNotNegotiateALPNWithDefaultRoute() throws Exception
+    {
+        k3po.finish();
+    }
+
+    /*
+     * only one route, for protocol2
+     * client sends ALPN w/ protocol3
+     * negotiates unsuccessfully with ALPN, so protocol == null => TLS ERROR
+     */
+    @Ignore("There no way to detect connection abort in k3po")
+    @Test
+    @Specification({
+            "${route}/server.alpn/controller",
+            "${client}/connection.not.established.with.wrong.alpn/client",
+            "${server}/connection.established/server" })
+    @ScriptProperty({
+            "newClientAcceptRef ${newServerConnectRef}",
+            "clientAccept \"nukleus://target/streams/tls#source\"" })
+    public void shouldNotNegotiateWithALPNAsProtocolMismatch() throws Exception
+    {
+        k3po.finish();
+    }
+
+    /*
+     * only one route, for protocol2
+     * client omits ALPN
+     * negotiates successfully without ALPN, but protocol != protocol2 => RESET transport
+     */
+    @Ignore("There no way to detect connection abort in k3po")
```

Comment:
File a ticket for this if it doesn't already exist, and refer to the ticket from this `@Ignore` annotation.